js/src/tests/test262/built-ins/String/prototype/replace/replaceValue-evaluation-order-regexp-object.js

```diff
@@ -23,8 +23,7 @@ replaceValue.toString = () => {
 
 let newString = "".replace("a", replaceValue);
 assert.sameValue(newString, "");
-// Taintfox: We change the semantics by calling toString/valueOf internally, so changed to 2
-assert.sameValue(calls, 2);
+assert.sameValue(calls, 1);
```

Comment:
How does this work now? Is it because we are checking the taint information before making calls?

Reply:
The Strings are untainted in this test, so we do not need to mess with the taint flow. Before, we created the TaintOperation, which in turn called toString() on the replaceValue() to record argument values for the taint flow.

Due to the check whether the thing we call replace on is tainted, we never create the TaintOperation and, consequently, do not call toString() anymore.

If the string were tainted, this side effect would be visible again, but this is something we probably can not avoid.

Reply:
Thanks for clarifying!